### PR TITLE
Switch to pnpm

### LINKED ISSUE #2197: Migrate yarn v2
Someone can take care and migrate to yarn v2, it should cut us time for development and fresh installs 

Why we should migrate -> https://yarnpkg.com/getting-started/qa#why-should-you-upgrade-to-yarn-modern

Note to @yangwao -- check deployments if works ok!
- on beta
- on production

### Ref
- https://yarnpkg.com/getting-started/migration

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### What's new?

- [x] PR closes #2197 
- [x] switch package manager to [pnpm](https://pnpm.io/)

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

tested in GitHub actions and netlify: https://github.com/preschian/nft-gallery/pull/1/checks
Screen Shot 2022-03-23 at 12 18 57 PM

### CODE REVIEW

**Comment by preschian** [comment]
seems like need to update netlify not to use yarn in the netlify dashboard. it should be able to deploy to netlify, take a look at this PR-test https://github.com/preschian/nft-gallery/pull/1

cc @yangwao @vikiival

**Comment by yangwao** [comment]
what will be build commands?

You can also please update it in, as we have mentioned yarn there, if it builds, I'm happy to change it over to pnpm. 
- Contributing.md
- https://github.com/kodadot/nft-gallery/blob/main/FIRST_TIME.md

**Comment by preschian** [comment]
noted, I forgot about the docs. Just realised after doing the https://github.com/kodadot/nft-gallery/pull/2644 issue 😅
that's why I switch the PR to draft again haha

this is my netlify config, just leave it blank
Screen Shot 2022-03-24 at 7 35 33 PM

**Comment by preschian** [comment]
suggestion from @yangwao 
- [ ] wrap long command in package.json
- [ ] build != generate
- [ ] test it in cloudflare pages

**Comment by yangwao** [comment]
I will also need to replace build commands on CF pages 
image

**Comment by preschian** [comment]
tested in:
- cloudflare pages: https://github.com/preschian/nft-gallery/pull/1#issuecomment-1078661057
	- skip npm install in cf: pass `NPM_FLAGS = --version` as env. [source](https://community.cloudflare.com/t/add-pnpm-to-pre-installed-cloudflare-pages-tools/288514/5)
- netlify: https://github.com/preschian/nft-gallery/pull/1#issuecomment-1075885821
	- skip npm install in netlify: leave it blank in build settings
- local development
	
improvements:

&nbsp; | yarn | pnpm
--- | --- | ---
Netlify Deployment | 5m <br /><br /> Screen Shot 2022-03-25 at 2 57 18 PM | **3m** <br /><br /> Screen Shot 2022-03-25 at 2 54 50 PM
Install Deps in Actions | [1m 25s](https://github.com/kodadot/nft-gallery/runs/5676422402?check_suite_focus=true#step:4:35) | [**28s**](https://github.com/kodadot/nft-gallery/runs/5687650302?check_suite_focus=true#step:6:467)

**Comment by yangwao** [comment]
> tested in:
> 
> * cloudflare pages: [switch to pnpm preschian/nft-gallery#1 (comment)](https://github.com/preschian/nft-gallery/pull/1#issuecomment-1078661057)
>   
>   * skip npm install in cf: pass `NPM_FLAGS = --version` as env. [source](https://community.cloudflare.com/t/add-pnpm-to-pre-installed-cloudflare-pages-tools/288514/5)
> * netlify: [switch to pnpm preschian/nft-gallery#1 (comment)](https://github.com/preschian/nft-gallery/pull/1#issuecomment-1075885821)
>   
>   * skip npm install in netlify: leave it blank in build settings
> * local development
> 
> improvements:
> 
>  	yarn	pnpm
> Netlify Deployment	5m
> 
> Screen Shot 2022-03-25 at 2 57 18 PM	**3m**
> 
> Screen Shot 2022-03-25 at 2 54 50 PM
> Install Deps in Actions	[1m 25s](https://github.com/kodadot/nft-gallery/runs/5676422402?check_suite_focus=true#step:4:35)	[**28s**](https://github.com/kodadot/nft-gallery/runs/5687650302?check_suite_focus=true#step:6:467)

impressive time savings.
Curious if you would be keen to write an article on how we've migrated from yarn to pnpm? 
Some people might find it interesting as we are still early! 
This morning found friends deciding between npm and yarn 😅.

We are usually publishing at https://medium.com/kodadot :)

**Comment by yangwao** [comment]
Once https://github.com/kodadot/nft-gallery/issues/2668 is resolved, we can check on deploying `pnpm`

**Comment by yangwao** [comment]
tried with empty credential or with npm run generate:pnpm and resulting in same errors 

maybe you have something different in your setup?

image

File: netlify.toml [hunk 1]
```diff
@@ -1,6 +1,10 @@
+[build.environment]
+  NODE_VERSION = "16"
+  NPM_FLAGS = "--version" # prevent Netlify npm install
+
 [build]
   # This will be your default build command.
-  command = "yarn generate"
+  command = "npm run generate:pnpm"
```
Comment (yangwao) [inline netlify.toml:7]:
```suggestion
  command = "npm i -g pnpm && npm run generate:pnpm"
```

**Review by yangwao — COMMENTED** [review]
maybe adding npm i -g pnpm would help

File: package.json [hunk 2]
```diff
@@ -23,13 +23,15 @@
...
     "generate": "nuxt generate",
+    "generate:pnpm": "npx pnpm i --store=node_modules/.pnpm-store && npx pnpm generate",
+    "start": "nuxt start",
     "lint": "eslint --ignore-path .gitignore --ext .js,.ts,.vue .",
     "lint:quiet": "eslint --quiet --ignore-path .gitignore --ext .js,.ts,.vue .",
     "lint:fix": "eslint --fix --quiet --ignore-path .gitignore --ext .js,.ts,.vue .",
     "test": "jest --verbose ./tests",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "preinstall": "npx only-allow pnpm"
```
Comment (yangwao) [inline package.json:34]:
```suggestion

```

**Comment by yangwao** [comment]
I'm confused @preschian 

Seems by this log it uses yarn? 
however, build instructions are
`npm i -g pnpm;npm run generate:pnpm`

https://app.netlify.com/sites/koda-nuxt/deploys/623ddcd0b5c58c0008c07402

**Comment by yangwao** [comment]
oh got hint 
image

**Comment by yangwao** [comment]
Oh apparently need merge something or remove file? 

image

**Comment by preschian** [comment]
@yangwao will update

**Comment by yangwao** [comment]
sending upper bound with something extra on top! :)

pay 300 usd 

thank you @preschian  for your amazing all-around cooperation!

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $300 @ 167.64 USD/KSM ~ 1.79 $KSM
🧗 [DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM](https://kusama.subscan.io/account/DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)
🔗 [0xded4881c8c32fc626530860209874b1851c788a020a0ea76ae33c9c00a86efaf](https://kusama.subscan.io/extrinsic/0xded4881c8c32fc626530860209874b1851c788a020a0ea76ae33c9c00a86efaf)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**